store/tikv/lock_resolver.go

```diff
@@ -807,7 +808,35 @@ func (lr *LockResolver) checkAllSecondaries(bo *Backoffer, l *Lock, status *TxnS
 		return nil, errors.Errorf("async commit recovery (sending CheckSecondaryLocks) finished with errors: %v", errs)
 	}
 
-	// TODO(nrc, cfzjywxk) schema lease check
+	// TODO check the schema lease considering related table ids.
```

Comment:
This TODO seems like a correctness issue. If the schema version changes but related tables don't change, here we roll back this transaction. However this transaction can pass the schema check on commit, and the user may receive success for this transaction.

Reply:
I think what's important is that the logic here need to be absolutely consistent with that in 2pc committing, so we have a  strict definition about when a transaction is said to be successfully committed: all keys are successfully prewritten, and, the schema must satisfy some conditions. Therefore I think it should be better if we can extract the logic here and that in 2pc.go into a single function to keep them perfectly consistent. Not sure if that's possible though.

Reply:
By the way, not very related to this PR though, I'm thinking that perhaps if the design about schema is changed in the future, the "strict definition" here may differ among different versions of TiDB. Perhaps we may need to disable async commit during upgrading for some future versions.

Reply:
We need to keep the check logic the same, it needs some work to refactor the underlying used interfaces. Seems it's better to make them all the same in one PR and avoid adding the new possible corrctness risks.